grafana-plugin/src/components/GForm/GForm.tsx

```diff
@@ -168,6 +168,7 @@ function renderFormControl(
   }
 }
 
+// TODO: make this generic to accept form data type
```

Comment:
@grafana/grafana-oncall-frontend is there an easy way to update a class component to be generic? (I'd prefer not to rewrite the component in this PR (as it makes use of `React.Comonent.forceUpdate`))

Reply:
You mean for the `data` to provide it under `<T>` or so - or some other params as well? Something like `class GForm<T> extends React.Component<GFormProps<T>, {}> {` where `T` will be the type for data.

I'm afraid for that is going to take some time until you figure out all the types passed from the other places such as schedules, maintenance and so on... Atm it's not very clear what their shape looks like. Feel free to open a tech debt/refactoring issue, we can work on it in the near future.

Reply:
Correct. I tried making the component generic using this approach and it _mostly_ works but it'd require a bit of refactoring on some parts of the existing component. I created [this issue](https://github.com/grafana/oncall/issues/3193) to refactor this in a separate PR 👍 

(I'll go ahead and remove my TODO comment here)